### ISSUE TITLE
Ignore line length containing urls

### ISSUE THREAD

**Issue by jeff1evesque** [markdownlint/markdownlint#128]
Line length, or the MD013 rule should be ignored, if a url is present within the corresponding line.  This should be similar to the `eslint` [max-len](http://eslint.org/docs/rules/max-len) implementation:

> ignoreUrls: Ignores lines that contain a URL.

**Note:** this issue is similar to #114.

**Comment by mivok** [comment]
The current implementation will ignore any long lines where there is no whitespace past the column in question. The intent for this was that you could have a long URL beginning before column 80 (for example) that would cause the line to become a long line without markdownlint flagging it. However, in this case, anything after the URL would have to be on a new line.

This covers most, but not all cases, and I agree that it would be useful to have an option to ignore any lines that contain a URL.

**Comment by dotLou** [comment]
I'm not sure about every implementation of markdown, but the way I got around this was to have the URLs later in the document like the following:

```markdown
For more info see [some website][some-website-url]
...
[some-website-url]: https://some-really-long.url.com/this-is-super-long-and-stuff
```

This has multiple benefits, IMO:
1. Lines are easier to read/edit in actual markdown mode because they are actually shorter
1. You can re-use the URL in multiple spots without duplication
1. The URL could change and you only have to do it in one spot, at the bottom of the document.
1. That last item won't trigger an error because the URL starts before the 80th character and is the only thing there
1. For really long URLs it makes following the document significantly easier.

**Comment by fpdotmonkey** [comment]
another case where this is relevant is codeblocks that involve URLS.  You may want to pass options in addition to the URL, so terminating the line with ` \` would make sense but include a space, e.g.

```sh
wget \
    -c https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.8/amd64/linux-headers-5.8.0-050800_5.8.0-050800.202008022230_all.deb \  # <== space causes MD013 warning
    -O headers.deb \
    #
```
